kyo-core/shared/src/main/scala/kyo/Retry.scala

```diff
@@ -42,15 +42,16 @@ object Retry:
             Frame
         ): A < (Async & Abort[E] & S) =
             Abort.run[E](v).map {
-                case Result.Success(r) => r
-                case error: Result.Error[?] =>
+                case Result.Success(value) => value
+                case result: Result.Failure[E] @unchecked =>
                     Clock.now.map { now =>
                         schedule.next(now).map { (delay, nextSchedule) =>
                             Async.delay(delay)(Retry[E](nextSchedule)(v))
                         }.getOrElse {
-                            Abort.get(error)
+                            Abort.get(result)
                         }
                     }
+                case Result.Panic(ex) => Abort.panic(ex)
```

Comment:
Might as well save an allocation 🙂 
```suggestion
                case panic: Result.Panic[?] => Abort.error(panic)
```